service/lib/dinstaller/dbus/storage/proposal.rb

```diff
@@ -33,7 +32,15 @@ class Proposal < ::DBus::Object
         INTERFACE = "org.opensuse.DInstaller.Storage.Proposal1"
         private_constant :INTERFACE
 
-        def initialize(logger, actions)
+        # Constructor
+        #
+        # @param backend [DInstaller::Storage::Proposal]
+        # @param logger [Logger]
+        # @param actions [DInstaller::DBus::Storage::Actions] D-Bus object representing the storage
+        #   actions to perform in the system. It is needed to raise signals when a new proposal is
+        #   calculated.
+        def initialize(backend, logger, actions)
```

Comment:
It would look better for me to inject the dependencies (`backend` and `actions`), and the logger at the end. Or we could even consider moving logger to a named argument. But it is just a matter of opinion, feel free to ignore this comment.

Reply:
Yes, using a named param is something I commented with @ancorgs. But actually we are not using name params in other places yet. So, again, I keep doing the same.

Reply:
The named param is something optional, my main point was about the arguments order.